tests/src/test/scala/ch/epfl/bluebrain/nexus/tests/kg/files/DiskStorageSpec.scala

```diff
@@ -33,34 +32,19 @@ class DiskStorageSpec extends StorageSpec with CopyFilesSpec {
     )
 
   override def createStorages(projectRef: String, storId: String, storName: String): IO[Assertion] = {
-    val payload       = jsonContentOf("kg/storages/disk.json", "id" -> storId)
     val storageId2    = s"${storId}2"
     val storage2Read  = s"$storName/read"
     val storage2Write = s"$storName/write"
-    val payload2      =
-      jsonContentOf("kg/storages/disk-perms.json", "id" -> storageId2, "read" -> storage2Read, "write" -> storage2Write)
+
+    val expectedStorage          = storageResponse(projectRef, storId, "resources/read", "files/write")
+    val expectedStorageWithPerms = storageResponse(projectRef, storageId2, storage2Read, storage2Write)
 
     for {
-      _ <- deltaClient.post[Json](s"/storages/$projectRef", payload, Coyote) { (_, response) =>
-             response.status shouldEqual StatusCodes.Created
-           }
-      _ <- deltaClient.get[Json](s"/storages/$projectRef/nxv:$storId", Coyote) { (json, response) =>
-             val expected = storageResponse(projectRef, storId, "resources/read", "files/write")
-             filterMetadataKeys(json) should equalIgnoreArrayOrder(expected)
-             response.status shouldEqual StatusCodes.OK
-           }
-      _ <- permissionDsl.addPermissions(
-             Permission(storName, "read"),
-             Permission(storName, "write")
-           )
-      _ <- deltaClient.post[Json](s"/storages/$projectRef", payload2, Coyote) { (_, response) =>
-             response.status shouldEqual StatusCodes.Created
-           }
-      _ <- deltaClient.get[Json](s"/storages/$projectRef/nxv:$storageId2", Coyote) { (json, response) =>
-             val expected = storageResponse(projectRef, storageId2, storage2Read, storage2Write)
-             filterMetadataKeys(json) should equalIgnoreArrayOrder(expected)
-             response.status shouldEqual StatusCodes.OK
-           }
+      _ <- storagesDsl.createDiskStorageDefaultPerms(storId, projectRef)
+      _ <- storagesDsl.checkStorageMetadata(projectRef, storId, expectedStorage)
```

Comment:
I thinnk this would be clearer as `storagesDsl.storageMetadata(project, storgage) shouldBe expectedStorage`, right now it's a bit unclear what the check is

Reply:
Another thing caused by the assertion passing pattern 😞 maybe it's worth refactoring all the DSLs?

Reply:
Ha well it would be nice but also probably an exercise in pain